### ISSUE TITLE
Allow selecting of files in Tree View to Edit or Download

### ISSUE THREAD

**Issue by shlake** [IQSS/dataverse#6691]
While in Tree view, it would be helpful if a user could select files from this view to download.

As a dataset owner, it would be helpful if one could select files and perform all the "Edit Files" functions.

**Comment by djbrooke** [comment]
Hey @shlake, we decided not to address providing more functionality (explore, edit, etc.) in the tree view because it would be very challenging, especially with any batch operations, but we can revisit the future. It's probably not something we'll get to soon. If we hyperlinked files we'd probably link to the file page instead of downloading.

**Comment by mheppler** [comment]
To Danny's point, the actions were a level of complexity we passed on trying to include in this initial phase of hierarchy support in the UI as those actions are available in both the table view and on the file page (through file name links in the component).

That said, looking at the PrimeFaces component showcase for the [Tree component](https://www.primefaces.org/showcase/ui/data/tree/selection.xhtml), there is a "selectionMode" attribute that we can use to add checkboxes for bulk actions like edit and download.

![Screen Shot 2020-02-27 at 1 04 38 PM](https://user-images.githubusercontent.com/687227/75473050-7ef6ec00-5962-11ea-99de-91281089acad.png)

**Comment by philippconzett** [comment]
Being able to select entire folders in tree view (or in another way), would be very useful. Currently, deleting or downloading all files in a given folder may be quite laborious when there are several tens or hundreds of files. We usually do this using several API calls including a bash file. But for common depositors, it would be good to make these kinds of operations available through the GUI.

**Comment by virgilejarrige** [comment]
Hi there! 
While asking researchers to try DVuploaders because they have datasets that have hundreds of little files, this feature request came almost immediately. So as Phil suggested it, I +1 this issue. :)
Take care!

**Comment by NessaTdoc** [comment]
Hi! 
Having the option to download specific folders in tree view would be really useful to easily locate and download files. 
Even if it's possible to download with the API, common users are turned off by this option and would prefer an easier option.
Thank you!

**Comment by pdurbin** [comment]
@NessaTdoc (and everyone) what if from the "Tree" view we linked to the new-ish "dirindex" view?

The dirindex view looks like this:

![Screen Shot 2022-09-01 at 11 40 31 AM](https://user-images.githubusercontent.com/21006/187956040-8ac6103d-f074-445a-b5ac-3307705693e4.png)

Still too geeky? 😄 

As you know the Tree index looks like this (for reference if we do want to add a link):

![Screen Shot 2022-09-01 at 11 40 27 AM](https://user-images.githubusercontent.com/21006/187956157-b0f60488-71cb-4d37-a45e-3306fc64e056.png)

You can play around with this dataset here:

- tree view: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/BOU7XR
- dirindex view: https://dataverse.harvard.edu/api/datasets/6443785/dirindex/?version=1.0&folder=QJE_35588_replication_package_submitted

**Comment by NessaTdoc** [comment]
@pdurbin via the dirindex the folders are now clickable, but can you download the whole folder withtout havign to click on individual files? 🤔 I seem to have the option to download only on files, not on the folder

**Comment by pdurbin** [comment]
@NessaTdoc you're right. dirindex only helps with individual files. I just wanted to make you and others aware of the feature as a possible workaround for downloading individual files. 😄 dirindex also doesn't help with editing files.

**Comment by DieuwertjeBloemen** [comment]
Hi,

At KU Leuven RDR, we've also been getting questions of users for a similar feature for larger datasets to make downloading a lot of file more structured. We are definitely willing to look into creating a PR for this. However, we thought of maybe developing a simplified version of this. Namely adding a download button for each folder instead of working with selection tick boxes. This because the tix boxes could be a lot more complicated to implement. So, what do you think of this idea? If you have other ideas, be sure to let us know as well as we're still in the brain storming phase.
Here is a preliminary mock-up of what we had in mind:

![Mock-up tree view folder download option](https://user-images.githubusercontent.com/92870663/225564857-24c11b95-e45e-4e2d-ba8b-a500873c0349.png)

**Comment by pdurbin** [comment]
@DieuwertjeBloemen a PR would be great! However, instead of the old JSF UI, how do you feel about contributing to the new React UI? That repo is over at https://github.com/IQSS/dataverse-frontend

To be honest, we're not quite ready for contributions yet 😅 but soon, hopefully!

**Comment by DieuwertjeBloemen** [comment]
@pdurbin I think that's definitely something that we might be interested in. I am just wondering what the estimated timing of the new UI is. Is there any idea/estimate on that? Because if it's relatively far off, then it might be worth doing both or if not, the opposite.

**Comment by pdurbin** [comment]
@DieuwertjeBloemen pretty far off, I'd say!

**Comment by DieuwertjeBloemen** [comment]
Then we're going to discuss it internally to see what the effort necessary is compared to the pay-off of doing it for the current UI. We'll let you know if we decide to move forward with it.

**Comment by DieuwertjeBloemen** [comment]
I think we will for now try to develop it for the current UI, as there is a pretty decent/urgent demand for it from our users. But we'll also keep in mind how to allow it to be reused in the new UI by possibly developing an API call that can do the same.
@ErykKul will likely be the person picking this up sometime soon.

**Comment by DieuwertjeBloemen** [comment]
@pdurbin I'm not sure what you mean by that. Do you mean that I create a PR with the full context and mock-up etc. like in the PR you linked or something else?

**Comment by pdurbin** [comment]
@DieuwertjeBloemen sorry, I dashed off that note without much context. Maybe I can give a quick demo of Storybook during the community call in an hour or so.

Another tool I've been playing with is Penpot (like Figma but open source). You can see a diagram I drew at https://github.com/IQSS/dataverse/issues/9150#issuecomment-1411175203

At a high level, I'm wondering if as an open source community we can do a little design, a little bit of mocking up our ideas. It should be faster than writing code. 😄 

I would think designs would be useful for either frontend, the old one or the new one.

You might already have some favorite design tools. I forget. Please let me know! 😄 

On a related note, please see also the excellent slides from @michbarsinai's talk "Draw More, Work Less"! 😄 https://www.mbarsinai.com/blog/2014/01/12/draw-more-work-less/

**Comment by cmbz** [comment]
To focus on the most important features and bugs,     we are closing issues created before 2020 (version 5.0)     that are not new feature requests with the label 'Type: Feature'.    

If you created this issue and you feel the team     should revisit this decision, please reopen the issue and leave a comment.

**Comment by philippconzett** [comment]
Hi all, I still think this feature request is relevant as it would make file download more efficient and user-friendly.

Thank you for reconsidering whether this issues should be closed! For some reason, I'm not able to reopen the issue.

**Comment by pdurbin** [comment]
@philippconzett only the creator of the issue (@shlake) can re-open it or someone with enough access (like me) so I'll go ahead and re-open it.

I will say, this is something we are unlikely to work on in JSF, the old UI that we will be throwing away. Please feel free to create a fresh issue in the React repo: https://github.com/IQSS/dataverse-frontend

**Comment by qqmyers** [comment]
FWIW: I think since this issue was created we added a presort by path option in the UI so files in subdirs can be grouped together.

**Comment by philippconzett** [comment]
Sorry, I missed that it was not me but @shlake who open this issue. I'd like to encourage @shlake to re-create it in the React repo.
